components/Table/Table.tsx

```diff
@@ -70,161 +70,163 @@ type TableProps = {
 	compactPagination?: boolean;
 };
 
-export const Table: FC<TableProps> = ({
-	columns = [],
-	columnsDeps = [],
-	data = [],
-	options = {},
-	noResultsMessage = null,
-	onTableRowClick = undefined,
-	palette = 'primary',
-	isLoading = false,
-	className,
-	showPagination = false,
-	pageSize = null,
-	hiddenColumns = [],
-	hideHeaders,
-	highlightRowsOnHover,
-	showShortList,
-	sortBy = [],
-	lastRef = null,
-	compactPagination = false,
-}) => {
-	const memoizedColumns = useMemo(
-		() => columns,
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-		columnsDeps
-	);
+export const Table: FC<TableProps> = memo(
+	({
+		columns = [],
+		columnsDeps = [],
+		data = [],
+		options = {},
+		noResultsMessage = null,
+		onTableRowClick = undefined,
+		palette = 'primary',
+		isLoading = false,
+		className,
+		showPagination = false,
+		pageSize = null,
+		hiddenColumns = [],
+		hideHeaders,
+		highlightRowsOnHover,
+		showShortList,
+		sortBy = [],
+		lastRef = null,
+		compactPagination = false,
+	}) => {
+		const memoizedColumns = useMemo(
+			() => columns,
+			// eslint-disable-next-line react-hooks/exhaustive-deps
+			columnsDeps
```

Comment:
Just wondering if this is necessary? Seems it's an extra thing for devs to understand / think about when using a table that wouldn't be very obvious, we should just probably be more careful with memoization of the column data we pass?

Reply:
I don't think it's actually necessary. It's been there for so long though that I'm a little skeptical. Do we save this for when we replace `react-table` with `@tanstack/react-table`? I can add it to the checklist.